app/templates/admin/services/upsert.html

```diff
@@ -0,0 +1,160 @@
+{% extends 'admin/base.html' %}
+{% set page_title='Services' %}
+
+{% block action_buttons %}
+  <a href="/admin/services" class="btn btn-link">Back to list</a>
+{% endblock %}
+
+{% block content %}
+  <div class="row">
+    <div class="col-8">
+      <form id="serviceForm" method="POST" enctype="multipart/form-data">
+        <div class="mb-3">
+          <label class="form-label">Name</label>
+          <input id="name" type="text" class="form-control" name="name" value="{{ service["name"] if service else '' }}">
+        </div>
+
+        <div class="mb-3">
+          <label class="form-label">Description</label>
+          <textarea id="description" name="description" class="form-control" rows="3">{{ service["description"] if service else '' }}</textarea>
+        </div>
+
+        <div class="mb-3">
+          <label class="form-label">Images</label>
+          <input name="images" multiple="multiple" class="form-control" type="file" id="images">
+        </div>
+
+        <div class="mb-3" id="priceRows">
+          <label class="form-label">
+            Price
+          </label>
+          <button id="addPriceRow" type="button" class="btn btn-sm btn-link float-end">
+            Add Price
+          </button>
+          {% if service  %}
+            {% for price in service["price"] %}
+              <div class="input-group mb-1">
+                <input value="{{ price[0] }}" class="price-row form-control" type="text" placeholder="Name..."/>
+                <span class="input-group-text" id="addon-wrapping"><i class="fas fa-dollar-sign"></i></span>
+                <input value="{{ price[1] }}" class="price-row form-control" type="text" placeholder="Price..."/>
+                <button type="button" class="price-del-row btn btn-danger"><i class="fas fa-trash"></i></button>
+              </div>
+            {% endfor %}
+          {% endif %}
+        </div>
+        <div class="d-grid gap-2">
+          <button class="btn btn-primary full" type="submit">Save</button>
+        </div>
+      </form>
+    </div>
+    <div class="col">
+      {% if service %}
+        <label class="form-label">
+          Uploaded Images
+        </label>
+        {% for image in service.images %}
+          <div class="mb-3">
+            <img src="/static/images/{{ image }}" class="img-thumbnail" title="{{ image }}"/>
+            <div class="d-grid">
+              <button data-img="{{ image }}" class="price-rm-img btn btn-sm btn-secondary full" type="submit">Remove</button>
+            </div>
+          </div>
+        {% endfor %}
+      {% endif %}
+    </div>
+  </div>
+  <br><br>
+{% endblock %}
+
+{% block js %}
+  <script>
+    const addService = (payload) => {
+      $.ajax({
```

Comment:
I'm not a fan of using AJAX request here? we can discuss it further of why.. can we just send the request body through the form without using AJAX?

Reply:
Yup, I prefer the normal form submit too, but I have to use Ajax because I design the add new price layout to be dynamic. I'm still looking a better way to collect prices data to send to the server.